client/src/components/related/ProductCard.jsx

```diff
@@ -1,65 +1,154 @@
+
+    if (!product) {
+      return <div className='rp-card rp-card-placeholder' >Loading...</div>;
+    }
+
+    console.log(`Rendering ${product.name} product card`);
+
+    const { checkCache, updateCache, productId, selectedProduct, selectProduct, type, action } = this.props;
+    let hoverText, actionClass, actionSymbol;
+
+
+    if (type === 'Outfit') {
       hoverText = `Remove ${product.name} from outfit`;
       actionClass = 'remove-from-outfit-symbol';
       actionSymbol = '+';
     } else {
       hoverText = `Compare ${selectedProduct.name} with ${product.name}`;
       actionClass = 'rp-compare-symbol';
-      actionSymbol = String.fromCharCode(9734);
+      actionSymbol = String.fromCharCode(9733);
     }
-  }
 
-  const maxTitleTextLength = 60;
-  // console.log(`Rendering card for ${product.name}`);
-  let trimmedTitle = `${product.name}: ${product.slogan.toLowerCase()}`;
-  if (trimmedTitle.length > maxTitleTextLength) {
-    const endIndex = Math.min(trimmedTitle.indexOf('.'), trimmedTitle.indexOf(' ', maxTitleTextLength - 10));
-    trimmedTitle = trimmedTitle.slice(0, endIndex) + '...';
-  }
+    const maxTitleTextLength = 60;
+    let trimmedTitle = `${product.name}: ${product.slogan.toLowerCase()}`;
+    if (trimmedTitle.length > maxTitleTextLength) {
+      const endIndex = Math.min(trimmedTitle.indexOf('.'), trimmedTitle.indexOf(' ', maxTitleTextLength - 10));
+      trimmedTitle = trimmedTitle.slice(0, endIndex) + '...';
+    }
 
-  let key = 0;
-  return type === 'placeholder' ? (
-    <div className='rp-card rp-card-placeholder' >{value}</div>
-  ) : (
-    <div className='rp-card' title={`Select ${product.name}`} onClick={ handleProductClick }>
-      <div className='rp-image-backer'>
-        <img src=''></img>
-        <div className={actionClass} title={hoverText} value={product.id} onClick={ handleActionButtonClick } >{actionSymbol}</div>
-      </div>
-      <div className='rp-info'>
-        <span className='rp-category'>{product.category.toUpperCase()}</span>
-        <span className='rp-title'>{trimmedTitle}</span>
-        <span className='rp-price' title={`Default price: ${product.default_price}`}>{'$' + product.default_price}</span>
-        <div className='rp-star-rating' title='Average customer review: 3.8 out of 5'>
-          {stars.map(star => (
-            <a key={key++}>{String.fromCharCode((star > 0) ? 9733 : 9734)}</a>
-          ))}
+    let key = 0;
+    return (
+      <div className='rp-card' title={`More details on ${product.name}`} onClick={ this.handleProductClick }>
+        <div className='rp-image-backer'>
+          <img src={this.state.productImageURL}></img>
+          <div className={actionClass} title={hoverText} value={productId} onClick={ this.handleActionButtonClick } >{actionSymbol}</div>
+        </div>
+        <div className='rp-info'>
+          <span className='rp-category' title={`${product.name} product category`}>{product.category.toUpperCase()}</span>
```

Comment:
I made a CSS class to change text to all uppercase. Maybe that is a better solution? Text case seems to be closely tied with style looking across the app overall, in a way that corresponds to what looks like a consistent header type, so it could be a decision that is changed globally in the future.

Reply:
I think the general rule is to do with CSS whatever can be done with CSS.